ansible_base/resource_registry/rest_client.py

```diff
@@ -17,7 +17,7 @@ class ResourceAPIClient:
     Client for Ansible services to interact with the service-index/ api
     """
 
-    def __init__(self, service_url: str, service_path: str, requests_auth_kwargs: dict, verify_https=True):
+    def __init__(self, service_url: str, service_path: str, requests_auth_kwargs: dict, verify_https=True, raise_if_bad_request: bool = False):
```

Comment:
Note: This flag is not ever set in this PR code. It is planned to be used by other code using this client, but is introduced here.

Reply:
It is used in this pr:

```
        resp = requests.request(**kwargs)
        if self.raise_if_bad_request:
            resp.raise_for_status()
        return resp
```

Reply:
That's a part of the implementation, I'm saying that nothing is using this in DAB. It's an added parameter of the client class for making requests. Nothing makes a request in DAB with that. Only something external is passing that and setting to the non-default value of `True`. It would be good to have a DAB test for this, I assume this is what the live_server type of tests were created for.